homeassistant/components/climate/econet.py

```diff
@@ -0,0 +1,234 @@
+
+    def __init__(self, water_heater):
+        """Initialize the water heater."""
+        self.water_heater = water_heater
+
+    @property
+    def name(self):
+        """Return the devices name."""
+        return self.water_heater.name
+
+    @property
+    def available(self):
+        """Return if the the device is online or not."""
+        return self.water_heater.is_connected
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_FAHRENHEIT
+
+    @property
+    def device_state_attributes(self):
+        """Return the optional state attributes."""
+        data = {}
+        vacations = self.water_heater.get_vacations()
+        if len(vacations) > 0:
+            data[ATTR_VACATION_START] = vacations[0].start_date
+            data[ATTR_VACATION_END] = vacations[0].end_date
+        data[ATTR_ON_VACATION] = self.water_heater.is_on_vacation
+        todays_usage = self.water_heater.total_usage_for_today
+        data[ATTR_TODAYS_ENERGY_USAGE] = todays_usage
+        data[ATTR_IN_USE] = self.water_heater.in_use
+
+        return data
+
+    @property
+    def current_operation(self):
+        """
+        Return current operation one of the following.
+
+        ["eco", "heat_pump",
+        "high_demand", "electric_only"]
+        """
+        current_op = ECONET_STATE_TO_HA.get(self.water_heater.mode)
+        if current_op is None:
+            current_op = STATE_UNKNOWN
+        return current_op
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        op_list = []
+        modes = self.water_heater.supported_modes
+        for mode in modes:
+            ha_mode = ECONET_STATE_TO_HA.get(mode)
+            if ha_mode is not None:
+                op_list.append(ha_mode)
+            else:
+                error = "Invalid operation mode mapping. " + mode + \
+                    " doesn't map. Please report this."
+                _LOGGER.error(error)
+        return op_list
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        self.water_heater.set_target_set_point(target_temp)
+
+    def set_operation_mode(self, operation_mode):
+        """Set operation mode."""
+        op_mode_to_set = HA_STATE_TO_ECONET.get(operation_mode)
+        self.water_heater.set_mode(op_mode_to_set)
+
+    def add_vacation(self, start, end):
+        """Add a vacation to this water heater."""
+        if not start:
+            start = datetime.datetime.now()
+        else:
+            start = datetime.datetime.fromtimestamp(start)
+        end = datetime.datetime.fromtimestamp(end)
+        self.water_heater.set_vacation_mode(start, end)
+
+    def update(self):
+        """Get the latest date."""
+        self.water_heater.update_state()
+
+    @property
+    def should_poll(self):
+        """Econet should always poll."""
+        return True
```

Comment:
This is the default, so you can remove this.

Reply:
:+1: